routes/user.js

```diff
@@ -0,0 +1,166 @@
+require("express-async-errors");
+const express = require("express");
+const { Booking } = require("../models/booking");
+const authz = require("../middleware/authz");
+const validateID = require("../middleware/validateID");
+const validateParamsIDwithToken = require("../middleware/validateParamsIDwithToken");
+
+const router = express.Router();
+
+router.post(
+  "/requestToBook/:userId",
+  [validateID, validateParamsIDwithToken(["userId"]), authz(["Client"])],
+  async (req, res) => {
+    //requestToBook/:userId?arrivalDate=YYYY-MM-DD&&departureDate=YYYY-MM-DD&&roomId=:roomId
+    if (
+      !req.query.arrivalDate ||
+      !req.query.departureDate ||
+      !req.query.roomId ||
+      !req.params.userId
+    )
+      return res.status(400).send(`Invalid Request`);
+
+    const checkIfBookingExicts = await Booking.findOne({
```

Comment:
Please fix any english typo. I would suggest for you to install extension **_Code Spell Checker_**.